src/data_info/services/trash.clj

```diff
@@ -50,6 +53,31 @@
     (throw+ {:error_code ERR_NOT_AUTHORIZED
              :paths (filterv #(home-matcher user %) paths)})))
 
+(defn- delete-paths-thread
+  [async-task-id]
+  (let [jargon-fn (fn [cm async-task update-fn]
+                    (update-fn "deleted paths" :begin)
+                    (let [{:keys [username data]} async-task]
+                      (log/info data)
+                      (log/info (:trash-paths data))
+                      (doseq [^String p (:paths data)]
+                        (update-fn p :begin-delete)
+                        ;;; Delete all of the tickets associated with the file.
+                        (let [path-tickets (mapv :ticket-id (ticket-ids-for-path cm (:username cm) p))]
+                          (doseq [path-ticket path-tickets]
+                            (delete-ticket cm (:username cm) path-ticket)))
+
+                        (update-fn p :deleted-tickets)
+
+                        ;;; If the file isn't already in the user's trash, move it there
+                        ;;; otherwise, do a hard delete.
+                        (if (contains? (:trash-paths data) (keyword p))
+                          (move-to-trash cm p ((:trash-paths data) (keyword p)) username :update-fn update-fn)
```

Comment:
both here and in restore, below, the calls to `keyword` are because when parsing the JSON it gets from the async-tasks service, clojure turns the keys into keywords.